jmclient/jmclient/wallet_rpc.py

```diff
@@ -0,0 +1,669 @@
+            # TODO: we only allow .jmdat files, and assume they
+            # are actually wallets; but we should validate these
+            # wallet files before returning them (though JM itself
+            # never puts any other kind of file in this directory,
+            # the user conceivably might).
+            wallets = os.listdir(wallet_dir)
+            wallets = [w for w in wallets if w.endswith("jmdat")]
+            return make_jmwalletd_response(request, wallets=wallets)
+
+        #route to get external address for deposit
+        @app.route('/wallet/<string:walletname>/address/new/<string:mixdepth>', methods=['GET'])
+        def getaddress(self, request, walletname, mixdepth):
+            self.check_cookie(request)
+            if not self.wallet_service:
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                raise InvalidRequestFormat()
+            mixdepth = int(mixdepth)
+            address = self.wallet_service.get_external_addr(mixdepth)
+            return make_jmwalletd_response(request, address=address)
+
+        @app.route('/wallet/<string:walletname>/address/timelock/new/<string:lockdate>', methods=['GET'])
+        def gettimelockaddress(self, request, walletname):
+            self.check_cookie(request)
+            if not self.wallet_service:
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                raise InvalidRequestFormat()
+            try:
+                timelockaddress = wallet_gettimelockaddress(self.wallet_service,
+                                                        lockdate)
+            except Exception as e:
+                return InvalidRequestFormat()
+            if timelockaddress == "":
+                return InvalidRequestFormat()
+            return make_jmwalletd_response(request, address=address)
+
+        def get_listutxos_response(self, utxos):
+            res = []
+            for k, v in utxos.items():
+                v["utxo"] = k
+                res.append(v)
+            return res
+
+        #route to list utxos
+        @app.route('/wallet/<string:walletname>/utxos',methods=['GET'])
+        def listutxos(self, request, walletname):
+            self.check_cookie(request)
+            if not self.wallet_service:
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                raise InvalidRequestFormat()
+            # note: the output of `showutxos` is already a string for CLI;
+            # but we return json:
+            utxos = json.loads(wallet_showutxos(self.wallet_service, False))
+            utxos_response = self.get_listutxos_response(utxos)
+            return make_jmwalletd_response(request, utxos=utxos_response)
+
+        #route to start a coinjoin transaction
+        @app.route('/wallet/<string:walletname>/taker/coinjoin',methods=['POST'])
+        def docoinjoin(self, request, walletname):
```

Comment:
Currently, this method does not check if a Taker is already running, we could do like with YG method and return `401 Service already started.`
Also, we could probably try to prevent starting as taker while the maker is running and vice versa.